### ISSUE TITLE
Integrate slots module with paras and new registrar

### ISSUE THREAD

**Issue by rphmeier** [paritytech/polkadot#1513]
depends on #1508 , #1509 

This should come with an update to the `trait Registrar` and `trait SwapAux` currently exported from the `slots` module to better reflect the functionality of the current system.

**Comment by rphmeier** [comment]
@shawntabrizi is this closed by #2166 ?

**Comment by shawntabrizi** [comment]
@rphmeier no, it could be put in here: https://github.com/paritytech/polkadot/pull/2181

But probably I will just do another separate PR.
